systems/frontier_user_guide.rst

```diff
@@ -3863,6 +3863,109 @@ If it is necessary to have bit-wise reproducible results from these libraries, i
+
+``MPICH_ABORT_ON_ERROR``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most users assume MPI will cause their code to exit if an error occurs.
+Setting this environment variable to ``1`` will cause MPI to abort and produce a core dump when an error occurs, which may help with debugging but will consume significant disk space at scale.
+
+Although often ignored, every MPI call returns an instructive error code and one can often recover descriptive error messages from checks.
+Setting ``MPICH_ABORT_ON_ERROR=0``, checking return codes, and outputting descriptive error messages is an extremely valuable practice for debugging MPI applications.
+
+``MPICH_GPU_SUPPORT_ENABLED``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setting this environment variable to ``1`` will enable GPU-aware MPI support in Cray MPICH, and is required to pass GPU buffers directly to MPI calls.
+The library `libmpi_gtl_hsa.so` must also be linked in, otherwise Cray MPICH will print an error message and exit immediately. 
+See :ref:`exposing-the-rocm-toolchain-to-your-programming-environment`  for more details.
+
+Using GPU-aware MPI is highly recommended on Frontier because the HPE Slingshot NICs are attached directly to the AMD MI250X accelerators.
+There is a small but measurable latency impact for enabling GPU-aware MPI with CPU buffers.
+Applications that do not use GPU buffers in MPI calls may want to leave this variable unset or set to ``0``, especially if they are sensitive to small message latency.
+
+``MPICH_ASYNC_PROGRESS``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setting this environment variable to ``1`` will spawn a thread dedicated to making progress on outstanding MPI communication and automatically increase the MPI thread level to MPI_THREAD_MULTIPLE.
+Applications that use one-sided MPI (eg, ``MPI_Put``, ``MPI_Get``) or non-blocking collectives (eg, ``MPI_Ialltoall``) will likely benefit from enabling this feature.
+Users of multi-threaded applications should consider having one fewer compute thread to leave a core free for this MPI progress thread.
+
+This feature is not typically beneficial for traditional two-sided communication patterns (eg, ``MPI_Send``, ``MPI_Recv``) and blocking collectives (eg, ``MPI_Alltoall``), and the requirement to use MPI_THREAD_MULTIPLE may introduce unnecessary overhead.
+
+``MPICH_RANK_REORDER_METHOD``, ``MPICH_RANK_REORDER_DISPLAY``, and ``MPICH_RANK_REORDER_FILE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Rather than relying on the default rank-to-node placement, Cray MPICH provides a flexible mechanism to customize the placement of MPI ranks on nodes.
+Three methods are implemented in Cray MPICH itself: 0 - round-robin, 1 -  packed, and 3 - folded. These methods are explained in detail in the `Cray MPICH User Guide <https://cpe.ext.hpe.com/docs/latest/mpt/mpich/index.html>`_, but briefly described as:
```

Comment:
The folded method is number 2, not 3.  Just a typo in this line, the text below shows it correctly as 2

Reply:
type fixed